syntax = "proto3";

package proverpkg;

message ZethNote {
    string aPK = 1;
    string value = 2;
    string rho = 3;
    string trapR0 = 4;
    string trapR1 = 5;
}

message JSInput {
    repeated string merkleNode = 1; // Merkle path to the commitment
    int64 address = 2;
    ZethNote note = 3;
    string spendingASK = 4;
    string nullifier = 5;
}

// All the data is given as hexadecimal strings
message HexadecimalPointBaseGroup1Affine {
    string xCoord = 1;
    string yCoord = 2;
}

message HexadecimalPointBaseGroup2Affine {
    string xC1Coord = 1;
    string xC0Coord = 2;
    string yC1Coord = 3;
    string yC0Coord = 4;
}
